plugins/wasi_nn/wasinnenv.h

```diff
@@ -369,6 +373,25 @@ struct WasiNNEnvironment :
   static PO::Option<std::string> NNRPCURI; // For RPC client mode
   std::shared_ptr<grpc::Channel> NNRPCChannel;
 #endif
+
+  const Host::WASI::Environ *getEnv() const noexcept {
+    auto *WasiModule = CurrentFrame->getWASIModule();
+    if (WasiModule != nullptr) {
+      return static_cast<const WasmEdge::Host::WasiModule *>(WasiModule)
+          ->getEnv();
+    }
+    return nullptr;
+  }
+
+  void setCurrentFrame(const Runtime::CallingFrame *Frame) noexcept {
+    CurrentFrame = Frame;
+  }
+  const Runtime::CallingFrame *getCurrentFrame() const noexcept {
+    return CurrentFrame;
+  }
+
+private:
+  const Runtime::CallingFrame *CurrentFrame = nullptr;
```

Comment:
Actually, I don't think it's a good choice to record the calling frame here.
The only information we need is the WASI module, aspecially the WASI environment.

Reply:
> How about we only store the Environ class? We can reset the Environ before each WASI-NN function starts.

Yes, I think this is a better choice.